httpx/_client.py

```diff
@@ -681,7 +687,7 @@ def _transport_for_url(self, url: URL) -> httpcore.SyncHTTPTransport:
         Returns the transport instance that should be used for a given URL.
         This will either be the standard connection pool, or a proxy.
         """
-        for pattern, transport in self._proxies.items():
+        for pattern, transport in self._mounts.items():
```

Comment:
So only the thing, why we introduce `self._mounts` is a lookup over the dict items.

Then why it's a dictionary? May we have a list with tuples there?